components/iam/pkg/server/oidc.go

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package server
```

Comment:
Let's move this into `iam/pkg/api/oidc` such that we have the handlers separated from the underlying server that runs the handlers.

This does a couple of things:
* Makes the OIDCClientRoute/Handler more portable
* Forces you to make the InitiateFlow etc methods public

Reply:
make sense